### ISSUE TITLE
Brave crashing with "brave_shields::AdBlockBaseService::OnDATFileDataReady()" when "Continue where you left off"launching using 

### ISSUE THREAD

**Issue by catacavaco** [brave/brave-browser#5309]
## Description 
Brave randomly crashes when started with the "Continue where you left off" option failing to restore the tabs previously open.
Brave then auto closes and when started up again, the tabs that should be restored from previous session are no more.
That includes pinned tabs.

## Steps to Reproduce
Since this does not happen every time the computer is started up, these are the steps to reproduce when it happens
   1. Computer is shut down, boot up to Windows 10
   2. Open Brave the first time, browser crashes and closes automatically
   3. Open Brave second time, tabs are gone

## Actual result:
Clean window of Brave with no tabs

## Expected result:
Tabs from previous session
or
Recovery tab with option to restore previously open tabs

## Reproduces how often: 
Intermittent issue

## Brave version (brave://version info)
Brave	0.66.100 Chromium: 75.0.3770.142 (Official Build) (64-bit) 
Revision	8d93477bd18154c67a70a67019cb8032382c55dd-refs/branch-heads/3770@{#1160}
OS	Windows 10 OS Version 1903 (Build 18362.239)

## Other Additional Information:

- Is the issue reproducible on the latest version of Chrome? No
Whenever Chrome (or Chromium) crashes, it displays a recovery tab where one can reopen tabs from the previous session

**Comment by kjozwiak** [comment]
@catacavaco when you run into the above crash, mind going into `brave://crashes` and pasting the `Crash Report ID` into the issue so one of the devs can take a look? Example of a submitted crash report:

<img width="688" alt="Screen Shot 2019-07-21 at 8 23 31 PM" src="https://user-images.githubusercontent.com/2602313/61599353-299d7400-abf6-11e9-931b-4d23eb22aed7.png">

Might be related to #4416 regarding sessions not restoring after certain crashes occur.

**Comment by catacavaco** [comment]
Probably all the same issue

Uploaded Crash Report ID 44038cdb28436c82 (Local Context: 8a328a8b-01e0-45aa-ab72-cae597df81fd)
Crash report captured on Saturday, July 20, 2019 at 12:09:53 PM, uploaded on Monday, July 22, 2019 at 5:51:49 PM

Uploaded Crash Report ID 4b21030b8ffd74f7 (Local Context: a28c0cf3-e5ee-4665-900b-f4073cac34ac)
Crash report captured on Thursday, July 11, 2019 at 5:47:13 PM, uploaded on Monday, July 22, 2019 at 5:51:49 PM

Uploaded Crash Report ID 6f38f0ea738b3f3f (Local Context: 365bbd11-2412-4fcc-9395-eda5acb91570)
Crash report captured on Tuesday, July 9, 2019 at 7:12:17 PM, uploaded on Monday, July 22, 2019 at 5:51:49 PM

Uploaded Crash Report ID 288175601b739ac4 (Local Context: 1e8f8028-ac6b-4304-993f-6fb7688882ab)
Crash report captured on Tuesday, July 2, 2019 at 7:32:46 PM, uploaded on Monday, July 22, 2019 at 5:51:49 PM

**Comment by kjozwiak** [comment]
Seems like it's crashing via:

```
brave_shields::AdBlockBaseService::OnDATFileDataReady() [ad_block_base_service.cc
``

Added links to the actual crashes so it's easier for whoever investigates:

* https://stats.brave.com/dashboard#crash/5d35db93c43734001f58507c
* https://stats.brave.com/dashboard#crash/5d35db930e0705001f64841f
* https://stats.brave.com/dashboard#crash/5d35db92c43734001f585074
* https://stats.brave.com/dashboard#crash/5d35db92c43734001f585070

**Comment by simonhong** [comment]
@kjozwiak I think this crash log is very similar with https://github.com/brave/brave-browser/issues/4764. And That crash seems disappeared with rust based adblock library since `0.68`.

**Comment by rebron** [comment]
Fixed in 0.68.

**Comment by catacavaco** [comment]
Downloaded beta version 0.68.121 Chromium: 76.0.3809.100 (Official Build) beta (64-bit) and indeed it is fixed, thanks for the help :)
